install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml

```diff
@@ -283,6 +283,7 @@ spec:
             - NET_ADMIN
             - SYS_MODULE
           privileged: true
+          {{- toYaml .Values.securityContext | trim | nindent 10 }}
```

Comment:
as cilium requires some securityContext such as capabilities, privileged to work. A small question about this value `.Values.securityContext`, I am thinking about doing below point:
- rename to `additionalSecurityContext` to make it explicit
- add a link https://docs.cilium.io/en/latest/operations/system_requirements/#privileges in helm docs

/cc @reviewer